src/AssemblerHttpServer-Alignments.cpp

```diff
@@ -1182,6 +1183,262 @@ void Assembler::computeAllAlignments(
+            "<br>";
+
+    // Write the form.
+    html <<
+        "<form>"
+        "<input type=submit value='Compute marker alignments'>"
+        "<br><br>"
+        "<table>"
+        "<tr>"
+        "<td>Number of reads to sample: "
+        "<td><input type=text name=samples required size=8 " <<
+        (samplesIsPresent ? "value="+to_string(samples) : "") <<
+        " title='Enter any number'>"
+        "<tr>"
+        "<td>Minimum number of raw bases in read (default=0): "
+        "<td><input type=text name=minLength size=8 " <<
+        (minLengthIsPresent ? "value="+to_string(minLength) : "") <<
+        " title='Enter any number'>"
+        "<tr>"
+        "<td>Maximum number of raw bases in read (default=inf): "
+        "<td><input type=text name=maxLength size=8 " <<
+        (maxLengthIsPresent ? "value="+to_string(maxLength) : "") <<
+        " title='Enter any number'>"
+        "<tr>"
+        "<td>Show alignment results "
+        "<td><input type=checkbox name=showAlignmentResults"
+        << (showAlignmentResults ? " checked=checked" : "") <<
+        ">"
+        "</table>";
+
+    renderEditableAlignmentConfig(
+            computeAllAlignmentsData.method,
+            computeAllAlignmentsData.maxSkip,
+            computeAllAlignmentsData.maxDrift,
+            computeAllAlignmentsData.maxMarkerFrequency,
+            computeAllAlignmentsData.minAlignedMarkerCount,
+            computeAllAlignmentsData.minAlignedFraction,
+            computeAllAlignmentsData.maxTrim,
+            computeAllAlignmentsData.matchScore,
+            computeAllAlignmentsData.mismatchScore,
+            computeAllAlignmentsData.gapScore,
+            computeAllAlignmentsData.downsamplingFactor,
+            computeAllAlignmentsData.bandExtend,
+            html
+    );
+
+    html << "</form>";
+
+    vector<OrientedReadId> sampledReads;
+
+    // If the user input is missing, stop here.
+    if(not samplesIsPresent) {
+        return;
+    }
+
+    // If the user doesn't care about filtering length, sample uniformly
+    if(not minLengthIsPresent and not maxLengthIsPresent) {
+        sampleReads(sampledReads, samples);
+    }
+    // Or else use any provided filters (defaults are used if only one is set)
+    else{
+        sampleReads(sampledReads, samples, minLength, maxLength);
+    }
+
+    // Initialize histograms
+    IterativeHistogram alignedFractionHistogram(0,1,20);
+    IterativeHistogram markerCountHistogram(0,3000,120);
+    IterativeHistogram nAlignmentsHistogram(0,200,20);
+
+    vector<pair<OrientedReadId, AlignmentInfo> > allAlignments;
+    vector<pair<OrientedReadId, AlignmentInfo> > allStoredAlignments;
+
+    for (auto& orientedReadId: sampledReads) {
+        // Vectors to contain markers sorted by kmerId.
+        vector<MarkerWithOrdinal> markers0SortedByKmerId;
+        vector<MarkerWithOrdinal> markers1SortedByKmerId;
+        getMarkersSortedByKmerId(orientedReadId, markers0SortedByKmerId);
```

Comment:
This is only necessary for alignment method 0.

Reply:
The page allows the user to pick any method

Reply:
Is it possible to skip the calls to `getMarkersSortedByKmerId` if the alignment method is not zero?

Reply:
Sure. This is all copied over from the standard alignments page, including the batchSize comment you made earlier. Should I update both pages?